components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/operator-finder/operator-finder.js

```diff
@@ -368,14 +435,34 @@ define(['jquery', 'lodash', 'log','remarkable', 'handlebar', 'designViewUtils','
                     if (isNotInstalledExtension(operatorExtension)) {
```

Comment:
For each operator in order to check whether its uninstalled extension aren't you creating the same array again again in the getUninstalledExtensionDetails method ? I think that is same for isPartialInstalledExtension , isInstalledExtension functions. Create the the arrays one time and use else this will be a huge process